source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -82,46 +94,33 @@ Tracing::SpanPtr Driver::startSpan(const Tracing::Config& config, Http::HeaderMa
   Tracer& tracer = *tls_->getTyped<TlsTracer>().tracer_;
   SpanPtr new_zipkin_span;
 
-  if (request_headers.OtSpanContext()) {
-    // Get the open-tracing span context.
-    // This header contains a span's parent-child relationships set by the downstream Envoy.
-    // The context built from this header allows the Zipkin tracer to
-    // properly set the span id and the parent span id.
-    SpanContext context;
-
-    context.populateFromString(request_headers.OtSpanContext()->value().c_str());
-
-    // Create either a child or a shared-context Zipkin span.
-    //
-    // An all-new child span will be started if the current context carries the SR annotation. In
-    // this case, we are dealing with an egress operation that causally succeeds a previous
-    // ingress operation. This envoy instance will be the the client-side of the new span, to which
-    // it will add the CS annotation.
-    //
-    // Differently, a shared-context span will be created if the current context carries the CS
-    // annotation. In this case, we are dealing with an ingress operation. This envoy instance,
-    // being at the receiving end, will add the SR annotation to the shared span context.
-
-    new_zipkin_span =
-        tracer.startSpan(config, request_headers.Host()->value().c_str(), start_time, context);
-  } else if (request_headers.XB3TraceId() && request_headers.XB3SpanId()) {
+  if (request_headers.XB3TraceId() && request_headers.XB3SpanId()) {
     uint64_t trace_id(0);
     uint64_t span_id(0);
     uint64_t parent_id(0);
+    bool sampled(true);
     if (!StringUtil::atoul(request_headers.XB3TraceId()->value().c_str(), trace_id, 16) ||
         !StringUtil::atoul(request_headers.XB3SpanId()->value().c_str(), span_id, 16) ||
         (request_headers.XB3ParentSpanId() &&
          !StringUtil::atoul(request_headers.XB3ParentSpanId()->value().c_str(), parent_id, 16))) {
       return Tracing::SpanPtr(new Tracing::NullSpan());
     }
 
-    SpanContext context(trace_id, span_id, parent_id);
+    if (request_headers.XB3Sampled()) {
+      sampled = !std::regex_match(request_headers.XB3Sampled()->value().c_str(), notSampledRegex());
```

Comment:
not seeing the benefit of using regex here. any advantage over checking both values in-line?

Reply:
Will change